<launch>
  <arg name="node_name" default="intensity_histogram" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />
  <arg name="queue_size" default="3" doc="Specigy queue_size of input image subscribers" />
  <arg name="histogram_size" default="64" doc="number of bins" />
  <arg name="reference_histogram" default="" doc="reference of histogram comparison" />


  <!-- intensity_histogram.cpp  -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="compare_histogram" >
    <remap from="image" to="$(arg image)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="queue_size" value="$(arg queue_size)" />
    <param name="histogram_size" value="$(arg histogram_size)" />
    <rosparam param="reference_histogram" subst_value="true">
      $(arg reference_histogram)
    </rosparam>
  </node>
</launch>
